lib/event.js

```diff
@@ -54,12 +54,35 @@ class EventModel extends BaseModel {
             endTime: config.endTime
         });
 
+        const findDuration = (start, end) => Math.round((new Date(end) - new Date(start)) / 1000);
+
         // Generate metrics
         const metrics = builds.map((b) => {
-            const { id, jobId, eventId, createTime, status, startTime, endTime } = b;
-            const duration = Math.round((new Date(endTime) - new Date(startTime)) / 1000);
+            const { id, jobId, eventId, createTime, status, startTime, endTime, stats } = b;
+            const duration = findDuration(startTime, endTime);
+            let queuedTime;
+            let imagePullTime;
+
+            // for backward compatibility, some old builds don't have stats field
+            if (stats) {
+                const { queueEnterTime, imagePullStartTime } = stats;
+
+                queuedTime = findDuration(queueEnterTime, imagePullStartTime);
```

Comment:
Should we make these metrics more consistent with what we're showing in the UI and just change this to `blockedTime = findDuration(createTime, imagePullStartTime)`? Or we can change UI to match. Not sure what makes more sense
https://github.com/screwdriver-cd/ui/blob/master/app/build/model.js#L81-L92

Reply:
Hmm yea, from the mock it looks like we includes that as `queuedTime`, which might be confusing when their builds are not really queued, but it was blocked. Should I change the key to `blockedTime`?

Reply:
To me, it makes more sense as `queuedTime`, since when it's blocked, it gets put back in the queue.
But I think @jithin1987 was worried people would think the queue was really slow if blocked jobs are shown as in the queue the whole time. 

What do you think makes more sense? Or should we just split to show both `queuedTime` and `blockedTime`?

Reply:
Let's get this merged and we can change the key later, so that it's not blocked by Jithin.